src/vs/workbench/contrib/chat/common/chatModel.ts

```diff
@@ -982,15 +982,20 @@ export interface ISerializableChatData3 extends Omit<ISerializableChatData2, 've
 	customTitle: string | undefined;
 }
 
+export interface ISerializableChatData4 extends Omit<ISerializableChatData3, 'version'> {
+	version: 4;
+	isToolsAgentModeEnabled: boolean;
```

Comment:
This is fine, but if you make this property optional, then you wouldn't have to make it a new chat data version